packages/bls-snap/src/createAccount.ts

```diff
@@ -2,22 +2,27 @@
 import { ethers } from 'ethers';
 import { BlsWalletWrapper, validateConfig } from 'bls-wallet-clients';
 
-import { ApiParams, CreateAccountRequestParams } from './types/snapApi';
+import { ApiParams } from './types/snapApi';
 import { ARBITRUM_GOERLI_NETWORK } from './utils/constants';
 import { upsertAccount } from './utils/snapUtils';
 
 export async function createAccount(params: ApiParams) {
   try {
-    const { state, mutex } = params;
+    const { state, mutex, wallet } = params;
     const netCfg = validateConfig(ARBITRUM_GOERLI_NETWORK.config);
 
     const provider = new ethers.providers.JsonRpcProvider(
-      'https://goerli-rollup.arbitrum.io/rpc',
+      ARBITRUM_GOERLI_NETWORK.rpcUrl,
+      { name: '', chainId: ARBITRUM_GOERLI_NETWORK.chainId },
     );
 
+    // const privateKey =
+    //   '0x0001020304050607080910111213141516171819202122232425262728293031';
+
     // 32 random bytes
-    const privateKey =
-      '0x0001020304050607080910111213141516171819202122232425262728293031';
+    const _arr = new Uint8Array(32);
+    crypto.getRandomValues(_arr);
+    const privateKey = ethers.utils.hexlify(_arr);
```

Comment:
Note that BLS keys live on different curve than ECDSA keys. This will work because we hash keys that are outside the curve to be compatible. But it would be better to use a native BLS private key here. Created https://github.com/web3well/bls-wallet/issues/442 to address.

Reply:
I see, but there is bigger problem with these two approaches. When user remove or re-install the snap (intentionally or not), the private key will be regenerated. So user will lose the account, it is not good. I'm working on an alternative solution which utilizes MetaMask api. I want to use seedphrase of MetaMask. There is api `snap_getBip44Entropy` for that. Take a look the [PR](https://github.com/aquiladev/bls-snap/pull/38)
As a `coinType` in `bip44` I use 9001 - Arbitrum
https://github.com/satoshilabs/slips/blob/master/slip-0044.md